diff --git a/fcrepo-server/src/main/java/org/fcrepo/server/management/DefaultManagement.java b/fcrepo-server/src/main/java/org/fcrepo/server/management/DefaultManagement.java
index adc1679..900e57d 100755
--- a/fcrepo-server/src/main/java/org/fcrepo/server/management/DefaultManagement.java
+++ b/fcrepo-server/src/main/java/org/fcrepo/server/management/DefaultManagement.java
@@ -748,11 +748,13 @@ public class DefaultManagement
             newds.DSLocation = dsLocation;
             newds.DSChecksumType = checksumType;
 
-            // validate reserved datastreams (type M and X)
-            ValidationUtility.validateReservedDatastream(PID.getInstance(pid),
-                                                         datastreamId,
-                                                         newds);
-
+            // validate reserved datastreams (type M and X) unless unchanged:
+            if (!newds.DSLocation.startsWith(DatastreamManagedContent.COPY_SCHEME) &&
+            		!newds.DSLocation.equals(orig.DSLocation)) {
+            	ValidationUtility.validateReservedDatastream(PID.getInstance(pid),
+                                                             datastreamId,
+                                                             newds);
+            }
 
             // next, add the datastream via the object writer
             w.addDatastream(newds, orig.DSVersionable);
