--- FilterFinalize.java	(revision 8867)
+++ FilterFinalize.java	(working copy)
@@ -111,8 +111,7 @@
             logger.debug(enter(method));
         }
         super.doThisSubclass(request, response);
-        request.lockWrapper();
-
+        
         if (REQUEST_ATTRIBUTE_INPUT_NAME != null) {
             Object testFedoraAuxSubjectAttributes =
                     request.getAttribute(REQUEST_ATTRIBUTE_INPUT_NAME);
@@ -181,6 +180,7 @@
                 + request.isUserInRole("eduPersonAffiliation"));
 
         request.setAttribute(DELIVERY_NAME, subjectAttributesMap);
+        request.lockWrapper();
         return false; // i.e., don't signal to terminate servlet filter chain
     }
 
