Index: src/java/fedora/server/security/servletfilters/xmluserfile/FedoraUsers.java
===================================================================
--- src/java/fedora/server/security/servletfilters/xmluserfile/FedoraUsers.java	(revision 6401)
+++ src/java/fedora/server/security/servletfilters/xmluserfile/FedoraUsers.java	(working copy)
@@ -5,6 +5,7 @@
 import java.io.IOException;
 import java.io.Serializable;
 import java.io.Writer;
+import java.net.URI;
 import java.util.ArrayList;
 import java.util.List;
 
@@ -21,7 +22,7 @@
 	private List<Role> roles;
 	private List<User> users;
 	
-	public static File fedoraUsersXML = new File(FEDORA_HOME + File.separator + 
+	public static final File fedoraUsersXML = new File(FEDORA_HOME + File.separator + 
 			"server" + File.separator + "config" + File.separator + "fedora-users.xml");
 	
 	public FedoraUsers() {
@@ -30,10 +31,10 @@
 	}
 	
 	public static FedoraUsers getInstance() {
-		return getInstance(fedoraUsersXML.getAbsolutePath());
+		return getInstance(fedoraUsersXML.toURI());
 	}
 	
-	public static FedoraUsers getInstance(String fedoraUsersXML) {
+	public static FedoraUsers getInstance(URI fedoraUsersXML) {
 		FedoraUsers fu = null;
 		BeanReader reader = new BeanReader();
 		reader.getXMLIntrospector().getConfiguration().setAttributesForPrimitives(false);
@@ -41,7 +42,7 @@
 
 		try {
 			reader.registerMultiMapping(getBetwixtMapping());
-			fu = (FedoraUsers)reader.parse(fedoraUsersXML);	
+			fu = (FedoraUsers)reader.parse(fedoraUsersXML.toString());	
 		} catch (IntrospectionException e) {
 			// TODO Auto-generated catch block
 			e.printStackTrace();


 	  	 
