--- code/old/ValidationUtility.java	2009-03-25 12:33:30.716680000 -0400
+++ code/new/ValidationUtility.java	2009-03-25 12:28:17.852782000 -0400
@@ -53,9 +53,9 @@
             throws ValidationException {
         try {
             URL candidate = new URL(url);
-            if (candidate.getProtocol().equals("file")) {
-                throw new ValidationException("Malformed URL: invalid protocol: " + url);
-            }
+//            if (candidate.getProtocol().equals("file")) {
+//                throw new ValidationException("Malformed URL: invalid protocol: " + url);
+//            }
         } catch (MalformedURLException e) {
             if (url.startsWith(DatastreamManagedContent.UPLOADED_SCHEME)) {
                 return;
