Index: fcrepo-server/src/main/java/org/fcrepo/server/storage/DefaultExternalContentManager.java
===================================================================
--- fcrepo-server/src/main/java/org/fcrepo/server/storage/DefaultExternalContentManager.java	(revision 8765)
+++ fcrepo-server/src/main/java/org/fcrepo/server/storage/DefaultExternalContentManager.java	(working copy)
@@ -175,6 +175,9 @@
                                                     knownMimeType);
             Property[] headerArray =
                     toPropertyArray(response.getResponseHeaders());
+            if (mimeType == null || mimeType.equals("")) {
+                mimeType = DEFAULT_MIMETYPE;
+            }
             return new MIMETypedStream(mimeType, response, headerArray);
         } catch (Exception e) {
             throw new GeneralException("Error getting " + url, e);
