diff -rupN fcrepo-src-3.4.2/fcrepo-server/src/main/java/org/fcrepo/server/storage/DefaultDOManager.java my-fcrepo-src-3.4.2/fcrepo-server/src/main/java/org/fcrepo/server/storage/DefaultDOManager.java
--- fcrepo-src-3.4.2/fcrepo-server/src/main/java/org/fcrepo/server/storage/DefaultDOManager.java	2011-05-17 16:00:44.000000000 +0200
+++ my-fcrepo-src-3.4.2/fcrepo-server/src/main/java/org/fcrepo/server/storage/DefaultDOManager.java	2011-05-24 14:37:05.756550993 +0200
@@ -1570,7 +1570,7 @@ public class DefaultDOManager
                     }
                 } catch (SQLException sqle) {
                     throw new StorageDeviceException("Error creating replication job: "
-                            + sqle.getMessage());
+                            + sqle.getMessage(), sqle);
                 } finally {
                     try {
                         if (results != null) {
@@ -1584,7 +1584,7 @@ public class DefaultDOManager
                         }
                     } catch (SQLException sqle) {
                         throw new StorageDeviceException("Unexpected error from SQL database: "
-                                + sqle.getMessage());
+                                + sqle.getMessage(), sqle);
                     } finally {
                         results = null;
                         s = null;
diff -rupN fcrepo-src-3.4.2/fcrepo-server/src/main/resources/dbspec/server/org/fcrepo/server/storage/resources/DefaultDOManager.dbspec my-fcrepo-src-3.4.2/fcrepo-server/src/main/resources/dbspec/server/org/fcrepo/server/storage/resources/DefaultDOManager.dbspec
--- fcrepo-src-3.4.2/fcrepo-server/src/main/resources/dbspec/server/org/fcrepo/server/storage/resources/DefaultDOManager.dbspec	2011-05-17 16:00:44.000000000 +0200
+++ my-fcrepo-src-3.4.2/fcrepo-server/src/main/resources/dbspec/server/org/fcrepo/server/storage/resources/DefaultDOManager.dbspec	2011-05-25 09:58:42.384699827 +0200
@@ -28,7 +28,7 @@
 		<column name="doPID" type="varchar(64)" notNull="true" binary="true">
 			<comment>The PID of the object</comment>
 		</column>
-		<column name="systemVersion" type="smallint(6)" notNull="true" default="0">
+		<column name="systemVersion" type="int(11)" notNull="true" default="0">
 			<comment>The system version of the object.  This starts at zero on
                initial creation or import, and is subsequently incremented by
                one each time a change is committed to the definitive store.
diff -rupN fcrepo-src-3.4.2/fcrepo-server/src/test/java/org/fcrepo/server/utilities/DerbyDDLConverterIntegrationTest.java my-fcrepo-src-3.4.2/fcrepo-server/src/test/java/org/fcrepo/server/utilities/DerbyDDLConverterIntegrationTest.java
--- fcrepo-src-3.4.2/fcrepo-server/src/test/java/org/fcrepo/server/utilities/DerbyDDLConverterIntegrationTest.java	2011-05-17 16:00:44.000000000 +0200
+++ my-fcrepo-src-3.4.2/fcrepo-server/src/test/java/org/fcrepo/server/utilities/DerbyDDLConverterIntegrationTest.java	2011-05-26 09:04:09.404043932 +0200
@@ -47,7 +47,7 @@ public class DerbyDDLConverterIntegratio
 
     private final String DO_REGISTRY_TABLE_SPEC =
             "CREATE TABLE doRegistry (\n  " + "doPID VARCHAR(64) NOT NULL,\n  "
-            + "systemVersion SMALLINT NOT NULL DEFAULT 0,\n  "
+            + "systemVersion INT NOT NULL DEFAULT 0,\n  "
             + "ownerId VARCHAR(64),\n  "
             + "objectState VARCHAR(1) NOT NULL DEFAULT 'A',\n  "
             + "label VARCHAR(255) DEFAULT '',\n  "
