--- fcrepo-src-3.3/fcrepo-server/src/main/java/fedora/server/oai/FedoraOAIProvider.java	2009-11-27 18:14:08.000000000 +0100
+++ my-fcrepo-src-3.3/fcrepo-server/src/main/java/fedora/server/oai/FedoraOAIProvider.java	2010-06-07 15:41:35.107326623 +0200
@@ -330,12 +330,12 @@ public class FedoraOAIProvider
         SimpleDateFormat formatter =
                 new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss");
         if (from != null) {
-            out.append(" dcmDate>='");
+            out.append(" mDate>='");
             out.append(formatter.format(from));
             out.append("'");
         }
         if (until != null) {
-            out.append(" dcmDate<='");
+            out.append(" mDate<='");
             out.append(formatter.format(until));
             out.append("'");
         }
