--- fedora-3.2.1/trunk/src/test/junit/fedora/server/journal/readerwriter/multifile/TestLockingFollowingJournalReader.java	2009-06-29 14:03:32.000000000 +0200
+++ my-fedora-3.2.1/trunk/src/test/junit/fedora/server/journal/readerwriter/multifile/TestLockingFollowingJournalReader.java	2010-06-11 14:31:36.514050055 +0200
@@ -448,7 +448,7 @@ public class TestLockingFollowingJournal
     }
 
     private File createTempDirectory(String name) {
-        File directory = new File(System.getProperty("java.io.tmpdir"), name);
+        File directory = new File(System.getProperty("java.io.tmpdir"), name + Long.toString(System.nanoTime()));
         directory.mkdir();
         cleanOutDirectory(directory);
         directory.deleteOnExit();
