--- fcrepo-3.3/trunk/fcrepo-server/src/test/java/fedora/server/journal/readerwriter/multifile/TestLockingFollowingJournalReader.java	2009-11-27 18:14:00.000000000 +0100
+++ my-fcrepo-3.3/trunk/fcrepo-server/src/test/java/fedora/server/journal/readerwriter/multifile/TestLockingFollowingJournalReader.java	2010-06-11 14:17:48.184014115 +0200
@@ -448,7 +448,7 @@ public class TestLockingFollowingJournal
     }
 
     private File createTempDirectory(String name) {
-        File directory = new File(System.getProperty("java.io.tmpdir"), name);
+        File directory = new File(System.getProperty("java.io.tmpdir"), name + Long.toString(System.nanoTime()));
         directory.mkdir();
         cleanOutDirectory(directory);
         directory.deleteOnExit();
