Index: fcrepo-server/src/main/java/fedora/server/journal/readerwriter/multicast/rmi/RmiTransportWriter.java
===================================================================
--- fcrepo-server/src/main/java/fedora/server/journal/readerwriter/multicast/rmi/RmiTransportWriter.java	(revision 8470)
+++ fcrepo-server/src/main/java/fedora/server/journal/readerwriter/multicast/rmi/RmiTransportWriter.java	(working copy)
@@ -35,6 +35,8 @@
 
     private long itemIndex;
 
+    private boolean m_closed = false;
+
     /**
      * When the writer is created, open the connection to the receiver.
      */
@@ -77,7 +79,10 @@
     @Override
     public void close() throws IOException {
         try {
-            receiver.closeFile();
+            if (!m_closed) {
+                receiver.closeFile();
+                m_closed= true;
+            }
         } catch (JournalException e) {
             IOException wrapper = new IOException();
             wrapper.initCause(e);
Index: fcrepo-server/src/main/java/fedora/server/journal/readerwriter/multicast/rmi/RmiTransport.java
===================================================================
--- fcrepo-server/src/main/java/fedora/server/journal/readerwriter/multicast/rmi/RmiTransport.java	(revision 8470)
+++ fcrepo-server/src/main/java/fedora/server/journal/readerwriter/multicast/rmi/RmiTransport.java	(working copy)
@@ -6,6 +6,7 @@
 package fedora.server.journal.readerwriter.multicast.rmi;
 
 import java.io.BufferedWriter;
+import java.io.IOException;
 
 import java.net.InetAddress;
 import java.net.MalformedURLException;
@@ -196,6 +197,17 @@
             super.testStateChange(State.FILE_CLOSED);
             parent.writeDocumentTrailer(xmlWriter);
             xmlWriter.close();
+
+            /*
+             * SOME implementations of XMLWriter do not
+             * close all resources, so we need to close
+             * the rmi manually just in case.
+             */
+            try {
+                writer.close();
+            } catch (IOException e) {
+                throw new JournalException(e);
+            }
             super.setState(State.FILE_CLOSED);
         } catch (XMLStreamException e) {
             throw new JournalException(e);
