diff --git fcrepo-server/src/main/java/org/fcrepo/server/storage/translation/FOXMLDODeserializer.java fcrepo-server/src/main/java/org/fcrepo/server/storage/translation/FOXMLDODeserializer.java
index 9fac4c6..5f80078 100644
--- fcrepo-server/src/main/java/org/fcrepo/server/storage/translation/FOXMLDODeserializer.java
+++ fcrepo-server/src/main/java/org/fcrepo/server/storage/translation/FOXMLDODeserializer.java
@@ -405,7 +405,11 @@ public class FOXMLDODeserializer
                 if (m_dsVersId.equals("AUDIT.0")) {
                     m_gotAudit = true;
                 }
-                m_dsChecksumType = Datastream.CHECKSUMTYPE_DISABLED;
+                if (Datastream.autoChecksum){
+                	m_dsChecksumType = Datastream.getDefaultChecksumType();
+                }else{
+                	m_dsChecksumType = Datastream.CHECKSUMTYPE_DISABLED;
+                }
                 m_dsChecksum = Datastream.CHECKSUM_NONE;
             } else if (localName.equals("contentDigest")) {
                 m_dsChecksumType = grab(a, FOXML.uri, "TYPE");
