Index: src/main/java/fedora/server/rest/FedoraObjectResource.java
===================================================================
--- src/main/java/fedora/server/rest/FedoraObjectResource.java	(revision 8488)
+++ src/main/java/fedora/server/rest/FedoraObjectResource.java	(working copy)
@@ -160,7 +160,7 @@
             String format) {
 
         try {
-            Date asOfDateTime = parseDate(dateTime);
+            Date asOfDateTime = getAsOfDateTime(dateTime);
             Context context = getContext();
             ObjectProfile objProfile = apiAService.getObjectProfile(context, pid, asOfDateTime);
             String xml = getSerializer(context).objectProfileToXML(objProfile, asOfDateTime);
Index: src/main/java/fedora/server/rest/BaseRestResource.java
===================================================================
--- src/main/java/fedora/server/rest/BaseRestResource.java	(revision 8488)
+++ src/main/java/fedora/server/rest/BaseRestResource.java	(working copy)
@@ -174,4 +174,29 @@
         }
         return date;
     }
+	
+	public static Date parseMementoDatetime(String acceptDatetime) throws IllegalArgumentException {
+		Date date = null;
+		
+		if(acceptDatetime != null) {
+			date = parseDate(acceptDatetime.trim().substring(acceptDatetime.indexOf("{") + 1, acceptDatetime.indexOf("}")));
+		}
+		
+		return date;
+	}
+	
+	public Date getAsOfDateTime(String dTime) {
+		Date date = null;
+		
+		if(dTime != null) {
+			date = parseDate(dTime);	
+		} 
+		
+		String acceptDatetime = headers.getRequestHeaders().getFirst("x-accept-datetime");
+		if(date == null && acceptDatetime != null) {
+			date = parseMementoDatetime(acceptDatetime);
+		}
+		
+		return date;
+	}
 }
Index: src/main/java/fedora/server/rest/MethodResource.java
===================================================================
--- src/main/java/fedora/server/rest/MethodResource.java	(revision 8488)
+++ src/main/java/fedora/server/rest/MethodResource.java	(working copy)
@@ -90,7 +90,7 @@
             @QueryParam(RestParam.AS_OF_DATE_TIME)
             String dTime) {
         try {
-            Date asOfDateTime = parseDate(dTime);
+            Date asOfDateTime = getAsOfDateTime(dTime);
             return buildResponse(apiAService.getDissemination(
                     getContext(),
                     pid,
@@ -106,7 +106,7 @@
 
     private Response getObjectMethodsForSDefImpl(String pid, String sDef, String dTime, String format) {
         try {
-            Date asOfDateTime = parseDate(dTime);
+            Date asOfDateTime = getAsOfDateTime(dTime);
             Context context = getContext();
             ObjectMethodsDef[] methodDefs = apiAService.listMethods(context, pid, asOfDateTime);
             String xml = getSerializer(context).objectMethodsToXml(methodDefs, pid, sDef, asOfDateTime);
Index: src/main/java/fedora/server/rest/DatastreamResource.java
===================================================================
--- src/main/java/fedora/server/rest/DatastreamResource.java	(revision 8488)
+++ src/main/java/fedora/server/rest/DatastreamResource.java	(working copy)
@@ -59,7 +59,6 @@
  */
 @Path("/{pid}/datastreams")
 public class DatastreamResource extends BaseRestResource {
-
     /**
      * Inquires upon all object Datastreams to obtain datastreams contained by a
      * digital object. This returns a set of datastream locations that represent
@@ -82,7 +81,7 @@
             String format) {
 
         try {
-            Date asOfDateTime = parseDate(dateTime);
+            Date asOfDateTime = getAsOfDateTime(dateTime);
             Context context = getContext();
             MediaType mime = RestHelper.getContentType(format);
             DatastreamDef[] dsDefs = apiAService.listDatastreams(context, pid, asOfDateTime);
@@ -100,7 +99,7 @@
             return handleException(ex);
         }
     }
-
+	
     /**
      * Invoke API-M.getDatastream(context, pid, dsID, asOfDateTime)
      *
@@ -122,7 +121,7 @@
                      @DefaultValue("false")
                      boolean validateChecksum) {
         try {
-            Date asOfDateTime = parseDate(dateTime);
+            Date asOfDateTime = getAsOfDateTime(dateTime);
             Context context = getContext();
             Datastream dsProfile = apiMService.getDatastream(context, pid, dsID, asOfDateTime);
 
@@ -131,7 +130,7 @@
                   "No datastream could be found. Either there is no datastream for " +
                   "the digital object \""+pid+"\" with datastream ID of \""+dsID+
                   "\"  OR  there are no datastreams that match the specified " +
-                  "date/time value of \""+dateTime+"\".").build();
+                  "date/time value of \""+asOfDateTime.toString()+"\".").build();
             }
 
             String xml = getSerializer(context).
@@ -175,9 +174,9 @@
                                                                         context,
                                                                         pid,
                                                                         dsID,
-                                                                        parseDate(dateTime));
+                                                                        getAsOfDateTime(dateTime));
             if (datastreamFilenameHelper != null) {
-                datastreamFilenameHelper.addContentDispositionHeader(context, pid, dsID, download, parseDate(dateTime), stream);
+                datastreamFilenameHelper.addContentDispositionHeader(context, pid, dsID, download, getAsOfDateTime(dateTime), stream);
 
             }
 
