--- fcrepo-3.3/trunk/fcrepo-server/src/test/java/fedora/utilities/TestZip.java	2010-06-23 16:58:05.624458563 +0200
+++ my-fcrepo-3.3/trunk/fcrepo-server/src/test/java/fedora/utilities/TestZip.java	2010-06-23 17:16:42.473243117 +0200
@@ -18,8 +18,8 @@ public class TestZip
     private final String TMP_DIR = System.getProperty("java.io.tmpdir");
 
     private final String ZIP_FILE = TMP_DIR + File.separator + "test" + System.nanoTime() + ".zip";
-
-    private final String TEST_DIR = TMP_DIR + File.separator + "test";
+    private final String TEST_STR = "test" + System.nanoTime();
+    private final String TEST_DIR = TMP_DIR + File.separator + TEST_STR;
 
     @Override
     protected void setUp() throws Exception {
@@ -52,7 +52,7 @@ public class TestZip
     }
 
     public void testZip() throws Exception {
-        File dir = new File(TMP_DIR + File.separator + "test");
+        File dir = new File(TMP_DIR + File.separator + TEST_STR);
         //Zip.zip(ZIP_FILE, TMP_DIR + File.separator + "test");
         Zip.zip(new File(ZIP_FILE), dir.listFiles());
     }
