Index: src/java/fedora/server/security/ResourceAttributeFinderModule.java
===================================================================
--- src/java/fedora/server/security/ResourceAttributeFinderModule.java	(revision 61)
+++ src/java/fedora/server/security/ResourceAttributeFinderModule.java	(revision 62)
@@ -102,6 +102,9 @@
                               Constants.DATASTREAM.FORMAT_URI.datatype);
             registerAttribute(Constants.DATASTREAM.LOCATION.uri,
                               Constants.DATASTREAM.LOCATION.datatype);
+            
+            registerAttribute(Constants.MODEL.HAS_MODEL.uri,
+            		StringAttribute.identifier);            
 
             registerSupportedDesignatorType(AttributeDesignator.RESOURCE_TARGET);
             setInstantiatedOk(true);
@@ -260,7 +263,7 @@
                     return null;
                 }
 
-                values = models.toArray(values);
+                values = models.toArray(new String[0]);
             }
 
             /*
